File: prefill/spec/prefill.spec.js
'use strict';
var prefill = require('../prefill');

describe('prefill', function(){
    it('should work for integers', function(){
        expect(prefill(3, 1)).toEqual([1,1,1]);
    });

    it('should work for strings', function(){
        expect(prefill(2, "abc")).toEqual(["abc", "abc"]);
    });

    it('should work recursively', function(){
        expect(prefill(3, prefill(2, '2d'))).toEqual([['2d','2d'],['2d','2d'],['2d','2d']]);
    });

    it('should through a TypeError if inputs are bad', function(){
        expect(prefill("xyz", 1)).toEqual(new TypeError('you idiot'));
    });

    it('should work if v is undefined', function(){
        expect(prefill(2)).toEqual([undefined, undefined]);
    });
});



/*
prefill(3,1) #=>[1,1,1]

prefill(2,"abc") #=> ['abc','abc']

prefill("1", 1) #=> [1]

prefill(3, prefill(2,'2d'))
#=> [['2d','2d'],['2d','2d'],['2d','2d']]

prefill("xyz", 1)
#=> throws TypeError with message "xyz is invalid"
*/

File: prefill/spec/prefill_spec.rb
require_relative "../prefill"

describe "prefill" do
    it "should work for integers" do
        expect(prefill(3, 1)).to eq([1,1,1])
    end

    it "should work for strings" do
        expect(prefill(2, "abc")).to eq(["abc", "abc"])
    end

    it 'should work recursively' do
        expect(prefill(3, prefill(2, '2d'))).to eq([['2d','2d'],['2d','2d'],['2d','2d']]);
    end

    it 'should through a TypeError if inputs are bad' do
        expect(prefill("xyz", 1)).to eq(new TypeError('you idiot'));
    end

    it 'should work if v is undefined' do
        expect(prefill(2)).to eq([nil, nil]);
    end
end
